pkg/controller/azdrivernode.go

```diff
@@ -82,6 +87,27 @@ func (r *reconcileAzDriverNode) Reconcile(ctx context.Context, request reconcile
 	return reconcile.Result{Requeue: true}, err
 }
 
+func (r *reconcileAzDriverNode) MarkDetachRequired(ctx context.Context, nodeName string) error {
```

Comment:
Seems like you are updating the status for all volume attachments in the cluster here. Don't you need to filter the attachments with node name and only update those that belong to that node.

Reply:
Oops I forgot to add a conditional clause. Good catch!